hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MulticastService.java

```diff
@@ -209,8 +217,10 @@ private JoinMessage receive() {
             try {
                 final byte[] data = datagramPacketReceive.getData();
                 final int offset = datagramPacketReceive.getOffset();
-                final BufferObjectDataInput input = node.getSerializationService().createObjectDataInput(data);
-                input.position(offset);
+                final int length = datagramPacketReceive.getLength();
+
+                final BufferObjectDataInput input = node.getSerializationService()
```

Comment:
We should possibly do some special casing here. This way, even if the processor is no-op, it will introduce an array copy.